### PR TITLE
Add core repo as a git submodule

### PR DESCRIPTION
**Pull request by vga91**
Created via `git submodule add  --name apoc-core -b git_remote_as_repo_core https://github.com/neo4j/apoc apoc-core`.
If the `git_remote_as_repo_core` is merged, we should need to change `git_remote_as_repo_core` with `dev` (and/or `5.0` ?)

### CODE REVIEW

**Comment by vga91** [comment]
There are some failing tests (it seems due to a wrong path), but they are all part of `core` repository. 
I could try to solve them, but maybe, in order not to weigh down the build, it would be better to skip them. Wdyt?

File: settings.gradle
Comment (ncordon) [inline settings.gradle:22]: `processor` does not have a publish task, so it fails if added here

**Review by ncordon — APPROVED** [review]
LGTM but we should wait for another review on this, since I've made small modifications

File: settings.gradle
Comment (ncordon) [inline settings.gradle:19]: Without these the code from `core`, `common` would be tested as well, but we don't want to do that from the `extended` repo

File: settings.gradle
Comment (Lojjs) [inline settings.gradle:22]: There is no 'processor:test' and also no mention of 'test-startup' which was included in the CI.yaml. Is that intentional?

File: readme.adoc
Comment (Lojjs) [inline readme.adoc:288]: Not sure if it is worth pointing out in the readme, but I had to run this with sudo because otherwise I got permission denied for `rm -r apoc-core`

File: .gitmodules
Comment (Lojjs) [inline .gitmodules:4]: Is this an existing branch on the core repo or how does it work?

File: extended/build.gradle
Comment (Lojjs) [inline extended/build.gradle:50]: Two questions on this:

1. What does the `grep -q \"^-\"` part do?
2. Why do we need `rm -r apoc-core && git submodule update --init apoc-core` both here and in the init script?

**Review by Lojjs — COMMENTED** [review]
As far as I can judge it looks OK, but I had some questions

File: extended/build.gradle
Comment (Lojjs) [reply extended/build.gradle:50]: @vga91 Nacho couldn't explain this to me, can you?

Comment (vga91) [reply extended/build.gradle:50]: @Lojjs 

1. The `git submodule status apoc-core`, initially it has an output like this:
```
-COMMIT_HASH apoc-core
```
where the first `-` sign indicates that the submodule is not initialized.
Therefore, with `grep -q \"^-\"`, I'll check whether the module is initialized (and in this case I'll do `rm -r apoc-core && git submodule update --init apoc-core`) or not (then `git submodule update --remote apoc-core` will be executed).
It's the simplest method I've found to automatically check if the module is initialized and execute the correct command.

2. Because I could execute `Load gradle changes` (e.g. via IntelliJ), and in this case this piece of code will be executed, and the init script would be useless.

And therefore, every time I want to update the core changes, it would be enough to do "Load gradle changes"

Otherwise, with the CI (or by just doing `git clone <repo>` and `./gradlew shadow`), this piece of code would not be executed, so we need to manually run` init.gradle` (via `./gradlew  --init-script init.gradle build/shadow/otherCommand`)

Comment (Lojjs) [reply extended/build.gradle:50]: Thanks for the explanation, now I understand :)

**Review by Lojjs — APPROVED** [review]
Looks good now! I'm not really sure about the CI test failures, but they feel unrelated, so perhaps you can just try to rerun the CI.

**Review by ncordon — APPROVED** [review]

**Pull request merged**
